types/src/on_chain_config.rs

```diff
@@ -7,9 +7,76 @@ use crate::{
     language_storage::{StructTag, TypeTag},
     transaction::SCRIPT_HASH_LENGTH,
 };
+use anyhow::{format_err, Result};
 use libra_crypto::HashValue;
 use move_core_types::identifier::Identifier;
-use serde::{Deserialize, Serialize};
+use serde::{de::DeserializeOwned, Deserialize, Serialize};
+use std::collections::HashMap;
+
+/// To register an on-chain config in Rust:
+/// 1. Implement the `OnChainConfig` trait for the Rust representation of the config
+/// 2. Add the config's `ConfigID` to `ON_CHAIN_CONFIG_REGISTRY`
+
+#[derive(Copy, Clone, Eq, Hash, PartialEq)]
+pub struct ConfigID(&'static str);
+
+impl ConfigID {
+    pub fn access_path(self) -> AccessPath {
+        access_path_for_config(Identifier::new(self.0).expect("failed to get Identifier"))
+    }
+}
+
+pub const ON_CHAIN_CONFIG_REGISTRY: &[ConfigID] = &[VMPublishingOption::CONFIG_ID];
+
+#[derive(Clone)]
+pub struct OnChainConfigPayload {
+    configs: HashMap<ConfigID, Option<Vec<u8>>>,
+}
+
+impl OnChainConfigPayload {
+    pub fn new(configs: HashMap<ConfigID, Option<Vec<u8>>>) -> Self {
+        Self { configs }
+    }
+
```

Comment:
Should we implement the logic for calculating deltas over here?

Reply:
I thought about this too.. but in the end decided to keep it out of OnChainConfigPayload because it seemed to me that the logic for determining what a delta is/publishing policy was more specific to state sync, while OnChainConfigPayload is just the container for sending data. And if we want to change this logic in the future, it might be better to keep the impl in state sync. Also this logic isn't being repeated anywhere as of now